### DISCUSSION TITLE
How do I use `shellHook` in the template?

### DISCUSSION THREAD

**Discussion by billksun** [srid/haskell-template#82] [Q&A] [answered]
I want to run some shell commands when entering the dev shell. I use to do this with a `shellHook` in a vanilla flake. How would I do the same in `haskell-template`?

Furthermore, I may want to run different shell commands depending on the OS platform, how would I do this?

**Comment by srid** [comment, marked as answer]
You can use `overrideAttrs` on the devshell that haskell-flake returns, eg.: https://github.com/Platonic-Systems/mission-control/blob/9acdaa469ebd3c2d6816f8a30c0c217a0da59fe2/nix/flake-module.nix#L84-L91

> Furthermore, I may want to run different shell commands depending on the OS platform, how would I do this?

Check on `system`. eg: https://github.com/srid/nixos-config/blob/cc0bd940c9ff628949ba6b2f65d73b80ffaca1e7/lib.nix#L77-L86
